Snippets/cosmosdb/cosmosdb_0/SharedTransactions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Threading.Tasks;
+using NServiceBus;
+
+class MyMessage { }
+
+class MyMessageHandler : IHandleMessages<MyMessage>
+{
+    ToDoActivity test1 = new ToDoActivity();
+    ToDoActivity test2 = new ToDoActivity();
+    ToDoActivity test3 = new ToDoActivity();
+
+    public class ToDoActivity
+    {
+        public string id { get; set; }
+    }
+
+    #region CosmosDBHandlerSharedTransaction
+    public Task Handle(MyMessage message, IMessageHandlerContext context)
+    {
+        //setup the items for the batch...
+
+        context.SynchronizedStorageSession.GetSharedTransactionalBatch()
+                .CreateItem<ToDoActivity>(test1)
+                .ReplaceItem<ToDoActivity>(test2.id, test2)
+                .UpsertItem<ToDoActivity>(test3)
+                .DeleteItem("reading");
```

Comment:
```suggestion
                .DeleteItem("/item/id");
```

Deletion would be done by specifying the item's ID.

Reply:
Well, the ID is just a string. The example here partially mirrors the Microsoft example for TransactionBatch https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.transactionalbatch?view=azure-dotnet#examples

Reply:
I just wonder if we should provide a better example than Cosmos doco does 😄